src/contentscript/overlay/root/components/OverlayTab/index.tsx

```diff
@@ -75,11 +75,17 @@ export const OverlayTab = (p: OverlayTabProps): ReactElement => {
 
     handleUpdateNotifications()
     EventBus.on('notifications_updated', handleUpdateNotifications)
-    EventBus.on('show_notification', handleUpdateNotifications)
+    browser.runtime.onMessage.addListener((message) => {
+      if (!message || !message.type) return
+
+      if (message.type === 'SHOW_NOTIFICATION') {
+        return handleUpdateNotifications()
+      }
+    })
 
     return () => {
       EventBus.off('notifications_updated', handleUpdateNotifications)
-      EventBus.off('show_notification', handleUpdateNotifications)
+      browser.runtime.onMessage.removeListener(handleUpdateNotifications)
```

Comment:
The functions in addListener and removeListener differs. The unsubscribing won't work.